src/base/dense/owl_base_dense_ndarray_generic.mli

```diff
@@ -317,6 +317,10 @@ val sum : ?axis:int -> (float, 'b) t -> (float, 'b) t
 val sum' : (float, 'b) t -> float
 (** Refer to :doc:`owl_dense_ndarray_generic` *)
 
+(* TODO: change float to 'a *)
```

Comment:
Please make this function generic, refer to my previous comment.

Reply:
I've just updated the `owl_sum_along`, `sum'`, and `sum_reduce` functions to make them generic in [this commit](https://github.com/owlbarn/owl/pull/224/commits/1c2b309a1f057cb6b463b9b594d48cfe24a84c3c). Please take a look to see if that's a correct way to do it.